packages/http-client-csharp/emitter/src/lib/client-model-builder.ts

```diff
@@ -126,43 +127,26 @@ export function createModel(sdkContext: SdkContext<NetEmitterOptions>): CodeMode
   }
 
   function fromSdkEndpointParameter(p: SdkEndpointParameter): InputParameter[] {
-    if (p.type.templateArguments.length === 0)
-      return [
-        {
-          Name: p.name,
-          NameInRequest: p.serializedName ?? p.name,
-          Type: fromSdkType(p.type, sdkContext, modelMap, enumMap),
-          Location: RequestLocation.Uri,
-          IsApiVersion: false,
-          IsResourceParameter: false,
-          IsContentType: false,
-          IsRequired: true,
-          IsEndpoint: true,
-          SkipUrlEncoding: false,
-          Explode: false,
-          Kind: InputOperationParameterKind.Client,
-          DefaultValue: {
-            Type: {
-              Kind: "string",
-              Name: "string",
-              CrossLanguageDefinitionId: "TypeSpec.string",
-            },
-            Value: p.type.serverUrl,
-          },
-        },
-      ];
+    // TODO: handle SdkUnionType
+    if (p.type.kind === "union") {
+      return fromSdkEndpointType(p.type.values[0] as SdkEndpointType);
+    } else {
+      return fromSdkEndpointType(p.type);
+    }
+  }
 
+  function fromSdkEndpointType(type: SdkEndpointType): InputParameter[] {
```

Comment:
we have a `throw` branch in `converter.ts` for Endpoint type.
Could we consolidate this method with that?

Reply:
or at least we should add a todo here.

Reply:
consolidated, we should throw for endpoint in `converter.ts` as well.